src/modules/task/internal/task-reader.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  GetAllTaskParams,
+  GetTaskParams,
+  Task,
+  TaskNotFoundError,
+  PaginationParams,
+  GetTaskByNameParams,
+  TaskWithNameNotFoundError,
+} from '../types';
+import TaskRepository from './store/task-repository';
+import TaskUtil from './task-util';
+
+export default class TaskReader {
+  public static async getTask(params: GetTaskParams): Promise<Task> {
+    const task = await TaskRepository.taskDB.findOne({
+      _id: params.taskId,
+      account: params.accountId,
+      active: true,
+    });
+    if (!task) {
+      throw new TaskNotFoundError(params.taskId);
+    }
+    return TaskUtil.convertTaskDBToTask(task);
+  }
+
+  public static async getTaskByName(params: GetTaskByNameParams): Promise<Task> {
+    const task = await TaskRepository.taskDB.findOne({
```

Comment:
Is the task name unique? We might want to return an array.

Reply:
task name is not unique field , but for a particular account, there cannot 2 tasks with same name. I think this makes sense as I cannot create a task for washing multiple times.